### PR TITLE
Add quarkus property to set the hibernate flushmode

### LINKED ISSUE #29604: Add quarkus property to set the hibernate flushmode
### Description

By default, the hibernate flushmode is set to AUTO.
It would be beneficial to have a quarkus property to set the hibernate flushmode to other flushmodes e.g. COMMIT in case you don't want the default flushmode. 

### Implementation ideas

The name of the property could be _quarkus.hibernate-orm.flushmode_.

### PR DESCRIPTION
**Pull request by zedbeit**
My attempt to fix #29604 @gsmet pls review.

### CODE REVIEW

**Comment by zedbeit** [comment]
@gastaldi can you pls review.

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java [hunk 1]
```diff
@@ -236,6 +236,14 @@ public class HibernateOrmConfigPersistenceUnit {
     @ConfigItem(defaultValue = "true")
     public boolean validateInDevMode;
 
+    /**
+     * Represents flushing strategies (AUTO, COMMIT, ALWAYS, MANUAL).
+     * The default value is AUTO.
+     */
+    @ConfigItem
+    @ConvertWith(TrimmedStringConverter.class)
+    public Optional<String> flushMode;
```
[the remark this answers is not in the document]
Comment (Beita05) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java:245]: I don't think there is

[the remark this answers is not in the document]
Comment (zedbeit) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java:245]: I don't think there is

[the remark this answers is not in the document]
Comment (gastaldi) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java:245]: https://docs.jboss.org/hibernate/orm/6.1/javadocs/org/hibernate/FlushMode.html

[the remark this answers is not in the document]
Comment (zedbeit) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java]: Made some changes now

**Comment by zedbeit** [comment]
@gsmet @gastaldi pls can you review

**Review by yrodiere — CHANGES_REQUESTED** [review]
Thanks for your submission.

Your PR includes completely unrelated changes (padding stuff). You will have to remove that.

Also, I'm not sure we even want to allow multiple flush modes, as that might requires changes in other places. @Sanne any opinion on that?

**Comment by Sanne** [comment]
I'm indeed a bit skeptical about this, I had just commented on https://github.com/quarkusio/quarkus/issues/29604#issuecomment-1380714695 (before seeing this, sorry).

Let's keep the discussion on the issue? N.B. I'm skeptical but I am not saying "no way" ..

**Comment by zedbeit** [comment]
@Sanne @yrodiere based on the discussion on #29604, if the decision is to continue with the change, I'd like to continue working on this.

**Comment by Sanne** [comment]
@zedbeit thanks that's very kind but I expect we will not want to expose this. In fact there are discussions in the Hibernate ORM project to deprecate and/or possibly remove  this property altogether.

I might change my mind depending on if someone on #29604 comes forward with a reasonable explanation about why having this is important, but this seems unlikely to happen - so I would suggest to not spend any time on this for now.

Thanks anyways!

**Pull request closed without merging**